webui/react/src/components/WorkspaceMemberAddModal.tsx

```diff
@@ -19,84 +21,68 @@ interface Props {
-      user?: User;
-    };
-  };
-}
-
 const WorkspaceMemberAddModalComponent: React.FC<Props> = ({
   addableUsersAndGroups,
   rolesAssignableToScope,
   onClose,
-  workspaceId,
+  workspace,
 }: Props) => {
   const idPrefix = useId();
-  const [selectedOption, setSelectedOption] = useState<UserOrGroup>();
+  const [filteredOption, setFilteredOption] = useState<UserOrGroup[]>([]);
   const [form] = Form.useForm<FormInputs>();
+  const ref = useRef<RefSelectProps>(null);
 
-  const handleFilter = useCallback((search: string, option?: SearchProp): boolean => {
-    if (!option) return false;
-    const label = option.label;
-    const regex = new RegExp(search, 'i');
+  useEffect(() => {
+    ref.current?.focus();
+  }, [ref]);
 
-    return (
-      (label.props.name && regex.test(label.props.name)) ||
-      (label.props.user?.username && regex.test(label.props.user?.username)) ||
-      (label.props.user?.displayName && regex.test(label.props.user?.displayName)) ||
-      false
-    );
-  }, []);
-
-  const handleSelect = useCallback(
+  const handleSearch = useCallback(
     (value: string) => {
-      const userOrGroup = addableUsersAndGroups.find((u) => {
-        if (isUser(u) && value.substring(0, 2) === 'u_') {
-          const user = u;
-          return user.id === Number(value.substring(2));
-        } else if (!isUser(u) && value.substring(0, 2) === 'g_') {
-          const group = u;
-          return group.groupId === Number(value.substring(2));
-        }
-      });
-      setSelectedOption(userOrGroup);
+      const regex = new RegExp(value, 'i');
+      setFilteredOption(
+        value
+          ? _.filter(addableUsersAndGroups, (o) =>
+              isUser(o)
+                ? (o.displayName && regex.test(o.displayName)) || regex.test(o.username) || false
+                : (o.name && regex.test(o.name)) || false,
+            )
+          : [],
+      );
     },
     [addableUsersAndGroups],
   );
 
   const handleSubmit = useCallback(async () => {
     const values = await form.validateFields();
     try {
-      if (values && selectedOption) {
-        isUser(selectedOption)
-          ? await assignRolesToUser([
-              {
-                roleIds: [values.roleId],
-                scopeWorkspaceId: workspaceId,
-                userId: Number(values.userOrGroupId.substring(2)),
-              },
-            ])
-          : await assignRolesToGroup({
-              groupId: Number(values.userOrGroupId.substring(2)),
-              roleIds: [values.roleId],
-              scopeWorkspaceId: workspaceId,
-            });
+      if (values) {
+        const userOrGroup = _.groupBy(values.userOrGroupId, (o) => o.substring(0, 2));
+        const groupPayload = _.map(userOrGroup['g_'], (o) => ({
+          groupId: Number(o.substring(2)),
+          roleIds: [values.roleId],
+          scopeWorkspaceId: workspace.id,
+        }));
+        groupPayload.length > 0 && (await assignRolesToGroup(groupPayload));
```

Comment:
nit: it's probably personal preference but i think if statement is more readable than `&&`, but up to you